hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionHealthProblem.java

```diff
@@ -0,0 +1,103 @@
+
+public class SubscriptionHealthProblem {
+    private final ProblemCode code;
+    private final String description;
+
+    @JsonCreator
+    private SubscriptionHealthProblem(@JsonProperty("code") ProblemCode code,
+                                      @JsonProperty("description") String description) {
+        this.code = code;
+        this.description = description;
+    }
+
+    public static SubscriptionHealthProblem lagging(long subscriptionLag) {
+        return new SubscriptionHealthProblem(
+                LAGGING,
+                format("Subscription lag is growing, current value is %d messages", subscriptionLag)
+        );
+    }
+
+    public static SubscriptionHealthProblem malfunctioning(double code5xxErrorsRate) {
+        return new SubscriptionHealthProblem(
+                MALFUNCTIONING,
+                format("Consuming service returns a lot of 5xx codes, currently %.0f 5xx/s", code5xxErrorsRate)
+        );
+    }
+
+    public static SubscriptionHealthProblem receivingMalformedMessages(double code4xxErrorsRate) {
+        return new SubscriptionHealthProblem(
+                RECEIVING_MALFORMED_MESSAGES,
+                format("Consuming service returns a lot of 4xx codes, currently %.0f 4xx/s", code4xxErrorsRate)
+        );
+    }
+
+    public static SubscriptionHealthProblem slow(double subscriptionRate, double topicRate) {
+        return new SubscriptionHealthProblem(
+                SLOW,
+                format("Consumption rate (%.0f RPS) is lower than topic production rate (%.0f RPS)", subscriptionRate, topicRate)
+        );
+    }
+
+    public static SubscriptionHealthProblem timingOut(double timeoutsRate) {
+        return new SubscriptionHealthProblem(
+                TIMING_OUT,
+                format("Consuming service times out a lot, currently %.0f timeouts/s", timeoutsRate)
+        );
+    }
+
+    public static SubscriptionHealthProblem unreachable(double otherErrorsRate) {
+        return new SubscriptionHealthProblem(
+                UNREACHABLE,
+                format("Unable to connect to consuming service instances, current rate is %.0f failures/s", otherErrorsRate)
+        );
+    }
+
+    public ProblemCode getCode() {
+        return code;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public String toString() {
+        return code.name();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(code, description);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        SubscriptionHealthProblem other = (SubscriptionHealthProblem) obj;
+        return Objects.equals(this.code, other.code)
+                && Objects.equals(this.description, other.description);
+    }
+
+    public enum ProblemCode {
```

Comment:
Can we move those status codes at the top of the class? I think it will be easier to see them.